test/aio5gc/msg/nas/handler/authenticationResponse.go

```diff
@@ -17,7 +17,20 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func AuthenticationResponse(nasMsg *nas.Message, gnb *context.GNBContext, ue *context.UEContext, amf *context.AMFContext) error {
+func AuthenticationResponse(nasMsg *nas.Message, gnb *context.GNBContext, ue *context.UEContext, fgc *context.Aio5gc) error {
```

Comment:
Is there a need to that in each function?
Can't you do it in the switch a single time?

Also doesn't the behavior of this function depends on the state of UE? Shouldn't this function behavior be handled in a callbacks as things can change depending on the state?

Same comments for the others handlers function.